clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java

```diff
@@ -158,9 +175,15 @@ public InetAddress currentAddress(String id) throws UnknownHostException {
      */
     public void disconnected(String id, long now) {
         NodeConnectionState nodeState = nodeState(id);
-        nodeState.state = ConnectionState.DISCONNECTED;
         nodeState.lastConnectAttemptMs = now;
-        updateReconnectBackoff(nodeState);
+        incrementReconnectBackoff(nodeState);
```

Comment:
the attempt number is incremented, but backoff value is not exactly incremented. I think we should leave it as `updateReconnectBackoff`.

Reply:
Make sense. Refactored.